src/services/findAllReferences.ts

```diff
@@ -709,10 +709,26 @@ namespace ts.FindAllReferences.Core {
         return references.length ? [{ definition: { type: DefinitionKind.Symbol, symbol }, references }] : emptyArray;
     }
 
+    /** As in a `readonly prop: any` or `constructor(readonly prop: any)`, not a `readonly any[]`. */
+    function isMemberModifyingReadonlyKeyword(node: Node): boolean {
+        return node.kind === SyntaxKind.ReadonlyKeyword && hasReadonlyModifier(node.parent);
+    }
+
     /** getReferencedSymbols for special node kinds. */
     function getReferencedSymbolsSpecial(node: Node, sourceFiles: ReadonlyArray<SourceFile>, cancellationToken: CancellationToken): SymbolAndEntries[] | undefined {
         if (isTypeKeyword(node.kind)) {
-            return getAllReferencesForKeyword(sourceFiles, node.kind, cancellationToken);
+            // A modifier readonly (like on a property declaration) is not special;
+            // a readonly type keyword (like `readonly string[]`) is.
+            if (isMemberModifyingReadonlyKeyword(node)) {

```

Comment:
I think instead of checking node.parent modifiers you should check if this is readonly operator on Type.
````ts
export interface TypeOperatorNode extends TypeNode {
        kind: SyntaxKind.TypeOperator;
        operator: SyntaxKind.KeyOfKeyword | SyntaxKind.UniqueKeyword | SyntaxKind.ReadonlyKeyword;
        type: TypeNode;
    }
```

Reply:
That works, but I feel like it’s a weird way to write the function `isReadonlyModifierForMemberDeclaration`—if the function name implies that we’re checking if it’s used as a modifier, I’d rather check that it _is_ a modifier than check that it’s _not_ a type operator.

Reply:
You could name it to be isReadonlyTypeOperator and that would work. I think that's what you want to allow so you would want to check that.